File: TypeScript/plus_one/66.test.ts
import plusOne from "./66";

interface tt {
  digits: number[];
  want: number[];
}

let tests: tt[] = [
  { digits: [1, 2, 3], want: [1, 2, 4] },
  { digits: [4, 3, 2, 1], want: [4, 3, 2, 2] },
  { digits: [9], want: [1, 0] },
];

describe("plus one", () => {
  tests.forEach((t, i) => {
    test(`case ${i + 1}`, () => {
      const result = plusOne(t.digits);
      expect(result).toEqual(t.want);
    });
  });
});

File: Go/plus_one/66_test.go
package plusone

import (
	"fmt"
	"testing"

	"github.com/stretchr/testify/assert"
)

func Test_PlusOne(t *testing.T) {
	tests := []struct {
		digits []int
		want   []int
	}{
		{digits: []int{1, 2, 3}, want: []int{1, 2, 4}},
		{digits: []int{4, 3, 2, 1}, want: []int{4, 3, 2, 2}},
		{digits: []int{9}, want: []int{1, 0}},
	}

	for i, tt := range tests {
		t.Run(fmt.Sprintf("case %d", i+1), func(t *testing.T) {
			result := plusOne(tt.digits)
			assert.Equal(t, tt.want, result)
		})
	}
}
